Feat/portfolio store mobx

matthew-garrett opened the pull request:
- Adds portfolio state zero data to portfolio store
- Updates root store with default network and currentNetworkClient view

jeffywu commented on packages/core-entities/src/models/views/YieldViews.ts:
```diff
@@ -570,9 +561,13 @@ export const YieldViews = (self: Instance<typeof NetworkModel>) => {
 
     const allListedVaultsData = getTokensByType('VaultShare')
     .filter((t) => t.vaultAddress && listedVaultAddresses.includes(t.vaultAddress))
```
I still don't understand why either of these filters are ncessary

jeffywu commented on packages/features/portfolio/src/containers/portfolio-state-zero/hooks/use-card-data.tsx:
```diff
@@ -13,27 +13,18 @@ import { PORTFOLIO_STATE_ZERO_OPTIONS, PRODUCTS } from '@notional-finance/util';
   activeToken: string,
-  tokenData: ProductGroupItem[],
-  productGroupData: ProductGroupData
+  tokenData: any[],
```
Do not use `any`

jeffywu commented on packages/core-entities/src/models/views/YieldViews.ts:
```diff
@@ -41,6 +39,10 @@ export type ProductGroupItem = {
   underlying?: TokenDefinition;
   collateralFactor: string;
   debtToken?: TokenDefinition;
+  vaultData?: {
```
Why does this need to be nested like this? The debtToken is returned twice and what is the purpose of returning the non-leveraged APY?

jeffywu commented on packages/features/portfolio/src/containers/portfolio-state-zero/hooks/use-card-data.tsx:
```diff
@@ -98,15 +90,9 @@ export const useCardData = (
-  const pointsVault = tokenData?.find(
-    ({ apy, token }) => apy?.pointMultiples && token.tokenType === 'VaultShare'
-  );
+  const leveragedNToken = tokenData[0];
```
I do not like using an array here, we should use named keys instead.

jeffywu commented on packages/features/portfolio/src/containers/portfolio-state-zero/hooks/use-network-token-data.tsx:
```diff
@@ -1,105 +1,67 @@
-  defaultSymbol: string;
-} => {
-  const currentNetworkStore = useCurrentNetworkStore();
+export const useNetworkTokenData = (selectedTabIndex: number) => {
```
It doesn't look right to me that the set methods are not in a useEffect

matthew-garrett replied:
Imo this is an advantage not having them in a useEffect. This ensure's that the setters are only called once and then the data is just read.

jeffywu commented on packages/core-entities/src/models/views/YieldViews.ts:
```diff
@@ -633,10 +611,12 @@ export const YieldViews = (self: Instance<typeof NetworkModel>) => {
-    getPortfolioStateZeroEarnData,
-    getPortfolioStateZeroBorrowData,
-    getPortfolioStateZeroLeveragedData
+    getFCashLendData,
```
Rename these function names so they are more descriptive:
`getAllNTokenYields`, `getAllFCashYields`, etc.

"Data" does not describe accurately what is being returned.

jeffywu commented on packages/shared/web/src/dashboard-view/hooks/use-dashboard-config.tsx:
```diff
@@ -167,6 +169,7 @@ export const useDashboardConfig = (routeKey: PRODUCTS) => {
     const label = v === 0 ? Network.arbitrum : Network.mainnet;
```
Can you fix the capitalization on `NetWork`?

jeffywu commented on packages/shared/web/src/dashboard-view/hooks/use-fixed-rate-grid.tsx:
```diff
@@ -27,10 +29,10 @@ export const useFixedRateGrid = (
-    ? fCashYieldData.filter(({ token }) => token.isFCashDebt)
-    : fCashYieldData.filter(({ token }) => !token.isFCashDebt);
+  const currentNetworkStore = useCurrentNetworkStore();
+  const fCashLend = currentNetworkStore.getFCashLendData();
```
Since this is not in a hook, you don't need to call both functions, you can just do:
`const yieldData = isBorrow ? getFCashDebtData() : getFCashLendData()`

jeffywu commented on packages/shared/web/src/dashboard-view/hooks/use-lend-borrow-list.tsx:
```diff
@@ -29,17 +31,22 @@ export const useLendBorrowList = (
-    isBorrow,
-  });
+
+  const yieldData = yieldDataKeys[product];
```
Same as above, why fetch all of the data in this hook only to use one of them?

jeffywu commented on packages/shared/web/src/dashboard-view/hooks/use-leverage-farming-grid.tsx:
```diff
@@ -33,35 +30,30 @@ export const useLeveragedFarmingGrid = (
+      // TODO: Replace with new MOBX account data when ready
       const profile = vaultHoldings.find(
-        (p) => p.vault.vaultAddress === vaultAddress
+        (p) => p.vault.vaultAddress === vaultData?.vaultAddress
```
Based on the types above, the vaultAddress key doesnt exist. The vaultAddress key is on the token object.

jeffywu commented on packages/shared/web/src/dashboard-view/hooks/use-liquidity-list.tsx:
```diff
@@ -14,17 +14,21 @@ import {
+  const leveragedYieldData = currentNetworkStore.getLeveragedNTokenData();
+  const variableYieldData = currentNetworkStore.getNTokenData();
+
   const yieldData =
```
Same again, why fetch more data than you need

jeffywu commented on packages/shared/web/src/dashboard-view/hooks/use-variable-rate-grid.tsx:
```diff
@@ -13,16 +13,20 @@ import {
   const isBorrow = product === PRODUCTS.BORROW_VARIABLE;
-  const tokenType = isBorrow ? 'PrimeDebt' : 'PrimeCash';
-  const yieldData = useNetworkTokens(network, tokenType, { isBorrow });
+  const yieldData = isBorrow ? variableDebt : variableLend;
```
Fetch less data.

jeffywu commented on packages/shared/web/src/dashboard-view/hooks/use-lend-borrow-list.tsx:
```diff
@@ -29,17 +31,22 @@ export const useLendBorrowList = (
+
+  const yieldData = yieldDataKeys[product];
 
   let listColumns: DataTableColumn[] = [
```
All of the `listColumns` in every one of these hooks is re-declared in every hook, which means it is created and destroyed on every render. Since it is used as a prop it should not trigger extra re-renders but as a pattern I think this adds unnecessary computation, lets move all of the column definitions out of the hooks and declare them as constants.

matthew-garrett replied:
Logic happens to remove columns based on data in the hooks. So that will still have to happen somewhere

matthew-garrett replied:
I do not see it as a performance issue

jeffywu replied:
Well in my opinion it also makes the code harder to read

jeffywu commented on packages/core-entities/src/models/views/YieldViews.ts:
```diff
@@ -587,6 +582,7 @@ export const YieldViews = (self: Instance<typeof NetworkModel>) => {
           underlying: t.underlying ? getTokenByID(t.underlying) : undefined,
           collateralFactor: getDebtOrCollateralFactor(t, false),
```
Remove the collateral factor from vaults

jeffywu commented on packages/core-entities/src/models/views/YieldViews.ts:
```diff
@@ -587,6 +582,7 @@ export const YieldViews = (self: Instance<typeof NetworkModel>) => {
           underlying: t.underlying ? getTokenByID(t.underlying) : undefined,
```
After making the change to look at getAllListedVaults, you can get the tvl and liquidity values from `t.vaultTVL`

jeffywu commented on packages/features/portfolio/src/containers/portfolio-state-zero/hooks/use-card-data.tsx:
```diff
@@ -13,27 +13,19 @@ import { PORTFOLIO_STATE_ZERO_OPTIONS, PRODUCTS } from '@notional-finance/util';
-  productGroupData: ProductGroupData
+  tokenData: any[],
+  productGroupData: StateZeroItemType | []
 ) => {
```
Do one giant if statement.

jeffywu commented on packages/notionable/src/stores/portfolio-store.ts:
```diff
@@ -18,36 +76,264 @@ const APYDataModel = types.model('APYData', {
+        const listedVaultAddresses = listedVaults.map(
+          (v) => v.vaultAddress
+        );
+        const allListedVaultsData = clientNetwork
```
This looks like a duplicate of the method in getVaultData?

matthew-garrett committed:
build: gets leveraged liquidity dashboard rendering

build: adds lev liquidity list data

build: moves state zero data functions to YieldViews

build: adds root store

build: simplifies yield view functions

build: addresses comments

build: adds farming data

build: breaks up yield view functions to be used in portfolio and dashboards

build: removes log

build: wip

build: moves state zero data from yield views to portfolio store

build: updates data based off of network change

build: removes unneeded commented code

build: adds types

build: addresses comments
